c error recived during the compilation

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: linux, Linux Mint 20.1
Processor: 4 cpus, 64bit, little endian, Intel(R) Core(TM) i5-6300U CPU @ 2.40GHz
CC version: cc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0

getwd: /home/vincent/Github/vlang-btc/vlang-btc.btcrpc
vmodules: /home/vincent/.vmodules
vroot: /home/vincent/Github/v
vexe: /home/vincent/Github/v/v
vexe mtime: 2021-09-18 11:42:37
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 396eede.209b159

Git version: git version 2.25.1
Git vroot status: weekly.2021.37-26-g209b1595 (8 commit(s) behind V master)
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 333c14de

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg parser/conf_file.v`
```v
module parser

import os

// Main object of the config parser
pub struct ConfParser {
	// The path of the file
	file_path      string            [required]
	mainet_cofig   map[string]string
	testnet_config map[string]string
	regtest_config map[string]string
	signet_config  map[string]string
mut:
	actual_section ConfSection = ConfSection.btc
	done           bool
}

// Enumerate type that it is used to
// set the type if section that it is supported
pub enum ConfSection {
	btc = 0
	testnet = 1
	regtest = 2
	signet = 3
}

pub fn new(file_path string) ?&ConfParser {
	return &ConfParser{
		file_path: file_path
	}
}

pub fn (mut instance ConfParser) parse() ? {
	lines := os.read_lines(instance.file_path) ?

	for line in lines {
		// it is a comment
		line_clean := line.replace(' ', '')
		if line_clean.starts_with('#') {
			continue
		} else if line_clean.starts_with('[') && line_clean.ends_with(']') {
			section := line_clean.substr(1, line.len - 1)
			println(section)
		} else if line_clean.contains('=') {
			key_val := line_clean.split('=')
			key := key_val[0]
			val := key_val[1]
			println('Key $key  with value $val')
		}
	}
}

// Function to make the check of some propriety
// that need to be inside the config file
//
// This function return true, and an empity message, or
// a message that tell us what is the error, and what propriety
// it is missed.
pub fn (instance &ConfParser) check(req []string, section ConfSection) (bool, string) {
	match section {
		.btc { return instance.check_in_map(req, instance.mainet_cofig) }
		.testnet { return instance.check_in_map(req, instance.testnet_config) }
		.regtest { return instance.check_in_map(req, instance.regtest_config) }
		.signet { return instance.check_in_map(req, instance.signet_config) }
	}
}

// private method put the in the configuration the right
// propriety.
fn (mut instance ConfParser) put_val(key string, val string) {
	match instance.actual_section {
		.btc { instance.mainet_cofig[key](val) }
		.testnet { instance.testnet_config[key](val) }
		.regtest { instance.regtest_config[key](val) }
		.signet { instance.signet_config[key](val) }
	}
}

// Take the requirements key in a string and check if all the propriety are in te 
fn (instance ConfParser) check_in_map(req []string, conf_map map[string]string) (bool, string) {
	for req_key in req {
		if req_key !in conf_map {
			return false, 'Key $req not in the config'
		}
	}
	return false, ''
}

```

**What did you expect to see?**

No error like C error. This should never happen.


**What did you see instead?**
```
v -shared -autofree .
==================
/tmp/v/vlang-btc.5478275669182742922.tmp.so.c:14857: error: cannot convert 'struct string' to 'void *'
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -2811,19 +2811,29 @@ pub fn (mut c Checker) fn_call(mut node ast.CallExpr) ast.Type {
 			elem_typ := c.table.get_type_symbol(info.elem_type)
 			if elem_typ.info is ast.FnType {
 				return elem_typ.info.func.return_type
+			} else {
+				c.error('cannot call the element of the array, it is not a function',
+					node.pos)
 			}
 		} else if sym.kind == .map {
 			info := sym.info as ast.Map
 			value_typ := c.table.get_type_symbol(info.value_type)
 			if value_typ.info is ast.FnType {
 				return value_typ.info.func.return_type
+			} else {
+				c.error('cannot call the value of the map, it is not a function', node.pos)
 			}
 		} else if sym.kind == .array_fixed {
 			info := sym.info as ast.ArrayFixed
 			elem_typ := c.table.get_type_symbol(info.elem_type)
 			if elem_typ.info is ast.FnType {
 				return elem_typ.info.func.return_type
+			} else {
+				c.error('cannot call the element of the array, it is not a function',
+					node.pos)
 			}
+		} else {
+			// TODO: assert? is this possible?
 		}
 		found = true
 		return ast.string_type

--- a/vlib/v/checker/tests/index_invalid_call.out
+++ b/vlib/v/checker/tests/index_invalid_call.out
@@ -1,0 +1,13 @@
+vlib/v/checker/tests/index_invalid_call.vv:4:20: error: cannot call the value of the map, it is not a function
+    2 | fn main() {
+    3 |     m := map[string]string
+    4 |     eprintln(m['abc']())
+      |                       ^
+    5 |     array := ["", "1"]
+    6 |     array[0]()
+vlib/v/checker/tests/index_invalid_call.vv:6:11: error: cannot call the element of the array, it is not a function
+    4 |     eprintln(m['abc']())
+    5 |     array := ["", "1"]
+    6 |     array[0]()
+      |              ^
+    7 | }

--- a/vlib/v/checker/tests/index_invalid_call.vv
+++ b/vlib/v/checker/tests/index_invalid_call.vv
@@ -1,0 +1,7 @@
+// fixes https://github.com/vlang/v/issues/11539 , copied example map test code from https://github.com/spytheman
+fn main() {
+	m := map[string]string
+	eprintln(m['abc']())
+	array := ["", "1"]
+	array[0]()
+}

